backend/utils/advanceQueue.js

```diff
@@ -9,22 +9,40 @@ async function advanceQueue(locationData) {
         return;
```

Comment:
Nit: We can remove this check since it will be inside the loop.

Otherwise, LGTM!